### PR TITLE
Fix XSS in text inputs

### PR DESCRIPTION
**Pull request by marvec**

### CODE REVIEW

File: src/app/view/perspectives/table/body/rows/cell-group/data-cell/table-data-cell.component.html [hunk 1]
```diff
@@ -11,14 +11,30 @@
...
+           class="text-pre overflow-hidden h-100 mx-1 d-flex align-items-center"
+>>>>>>> Stashed changes
            [title]="data.dataValue | formatDataValue | stripHtml:false"
            [ngClass]="data.attribute?.constraint | constraintClass"
            [class.invalid-value]="!(data.dataValue | isDataValueValid)"
            readonly data-test="table-data-input">
+        {{data.dataValue | previewDataValue}}
       </div>
+      <ng-template #richText>
+        <div [innerHTML]="data.dataValue | previewDataValue | safeHtml"
```
Comment (kubedo8) [inline src/app/view/perspectives/table/body/rows/cell-group/data-cell/table-data-cell.component.html:30]: What if user puts XSS text in Text constraint?

File: src/app/view/perspectives/table/body/rows/cell-group/data-cell/table-data-cell.component.html [hunk 2]
```diff
@@ -11,14 +11,30 @@
...
 
+<<<<<<< Updated upstream
     <ng-container *ngIf="!dataInputInfo.showDataInput">
       <div *ngIf="dataInputInfo.hasValue" class="text-pre overflow-hidden h-100 mx-1 align-middle"
            [innerHTML]="data.dataValue | previewDataValue | safeHtml"
+=======
+    <ng-container *ngIf="!dataInputInfo.showDataInput && dataInputInfo.hasValue">
+      <div *ngIf="data.attribute?.constraint?.type !== constraintType.Text; else richText"
+           class="text-pre overflow-hidden h-100 mx-1 d-flex align-items-center"
+>>>>>>> Stashed changes
```
Comment (kubedo8) [inline src/app/view/perspectives/table/body/rows/cell-group/data-cell/table-data-cell.component.html:22]: Stashed changes

File: src/app/view/perspectives/table/body/rows/cell-group/data-cell/table-data-cell.component.html (hunk 1 above)
Comment (marvec) [reply src/app/view/perspectives/table/body/rows/cell-group/data-cell/table-data-cell.component.html:30]: The text input component encodes HTML characters in their & variants and it is displayed properly. I have tested this.

File: src/app/view/perspectives/table/body/rows/cell-group/data-cell/table-data-cell.component.html (hunk 2 above)
Comment (marvec) [reply src/app/view/perspectives/table/body/rows/cell-group/data-cell/table-data-cell.component.html:22]: 🤦

**Pull request merged**